apps/web/screens/UniversitySearch/UniversitySearch.tsx

```diff
@@ -0,0 +1,1219 @@
+import {
+  Box,
+  ActionCategoryCard,
+  ContentBlock,
+  Icon,
+  Inline,
+  Tag,
+  Text,
+  CTAProps,
+  Button,
+  GridContainer,
+  GridColumn,
+  GridRow,
+  Stack,
+  Checkbox,
+  Accordion,
+  AccordionItem,
+  Pagination,
+  LinkV2,
+  Input,
+  VisuallyHidden,
+  ToastContainer,
+  toast,
+  Hidden,
+  Filter,
+  FilterMultiChoice,
+} from '@island.is/island-ui/core'
+import { Screen } from '@island.is/web/types'
+import { withMainLayout } from '@island.is/web/layouts/main'
+import { useEffect, useRef, useState } from 'react'
+import * as styles from './UniversitySearch.css'
+import { Comparison, ListViewCard } from '@island.is/web/components'
+import { useRouter } from 'next/router'
+import Fuse from 'fuse.js'
+import { SearchProducts } from '@island.is/web/utils/useUniversitySearch'
+import { useWindowSize } from '@island.is/web/hooks/useViewport'
+import { theme } from '@island.is/island-ui/theme'
+import {
+  GET_UNIVERSITY_GATEWAY_FILTERS,
+  GET_UNIVERSITY_GATEWAY_PROGRAM_LIST,
+  GET_UNIVERSITY_GATEWAY_UNIVERSITIES,
+} from '../queries/UniversityGateway'
+import {
+  GetNamespaceQuery,
+  GetNamespaceQueryVariables,
+  GetUniversityGatewayActiveProgramsQuery,
+  Program,
+  ProgramFilter,
+  University,
+} from '@island.is/web/graphql/schema'
+import { GET_NAMESPACE_QUERY } from '../queries'
+import { useNamespace } from '@island.is/web/hooks'
+import { TranslationDefaults } from './TranslationDefaults'
+import { CustomNextError } from '@island.is/web/units/errors'
+import getConfig from 'next/config'
+
+const { publicRuntimeConfig = {} } = getConfig() ?? {}
+
+const ITEMS_PER_PAGE = 8
+const NUMBER_OF_FILTERS = 6
+const MAX_SELECTED_COMPARISON = 3
+
+interface UniversitySearchProps {
+  data: Array<Program>
+  namespace: Record<string, string>
+  filterOptions: Array<ProgramFilter>
+  locale: string
+  universities: Array<University>
+}
+
+interface FilterProps {
+  degreeType: Array<string>
+  modeOfDelivery: Array<string>
+  universityId: Array<string>
+  durationInYears: Array<string>
+  startingSemesterSeason: Array<string>
+  // applications: Array<string>
+  // fieldOfStudy: Array<string>
+  // location: Array<string>
+  // tuition: Array<string>
+  // tags: Array<string>
+}
+
+const intialFilters = {
+  degreeType: [],
+  modeOfDelivery: [],
+  durationInYears: [],
+  universityId: [],
+  startingSemesterSeason: [],
+  // applications: [],
```

Comment:
Will this be added later. could we remove this commented code?

Reply:
it will be added soon, when we get more data from the universities

Reply:
Could you further elaborate on the purpose of this review then?

A: Should the review wait for these to be resolved or 
B: are we being asked to review what is already here with the stated intent of fullly reviewing changed files later?

Reply:
or
C: This is ready to be deployed and ready to be updated as well